File: programmers/기능개발.py
from collections import deque

def solution(progresses, speeds):
    arr = deque(progresses)
    speeds = deque(speeds)
    ans = []

    while arr and arr[0] < 100:
        arr = deque([a + s for a, s in zip(arr, speeds)])

        for i in range(len(arr)):
            if i == len(arr) - 1 and arr[i] >= 100:
                ans.append(i + 1)
            if arr[i] < 100:
                if i > 0:
                    ans.append(i)
                for _ in range(i):
                    arr.popleft()
                    speeds.popleft()
                break

    return ans

File: programmers/기능개발.js
// progresses 다 비울 까지 반복
// 1. speeds 순회하며 각 progresses에 추가
// 2. progresses[0]이 100을 넘을 때마다 shift, cnt++
// 3. cnt개 배포, cnt는 다시 0으로 초기화

function solution(progresses, speeds) {
    const ans = []
    let head = 0
    while (head < progresses.length) {
        let cnt = 0
        
        for (let i=head; i<progresses.length; i++) {
            progresses[i] += speeds[i]
        }
        
        for (let i=head; i<progresses.length; i++) {
            if (progresses[i] >= 100) {
                head++
                cnt++
            } else {
                break
            }
        }
        
        if (cnt > 0) {
            ans.push(cnt)
        }
    }
    
    return ans;
}